orcid-core/src/main/java/org/orcid/core/manager/v3/read_only/impl/AddressManagerReadOnlyImpl.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * =============================================================================
+ *
+ * ORCID (R) Open Source
+ * http://orcid.org
+ *
+ * Copyright (c) 2012-2014 ORCID, Inc.
+ * Licensed under an MIT-Style License (MIT)
+ * http://orcid.org/open-source-license
+ *
+ * This copyright and license information (including a link to the full license)
+ * shall be included in its entirety in all copies or substantial portion of
+ * the software.
+ *
+ * =============================================================================
+ */
+package org.orcid.core.manager.v3.read_only.impl;
+
+import java.util.List;
+
+import javax.annotation.Resource;
+
+import org.orcid.core.adapter.v3.JpaJaxbAddressAdapter;
+import org.orcid.core.manager.v3.read_only.AddressManagerReadOnly;
+import org.orcid.jaxb.model.v3.dev1.record.Address;
+import org.orcid.jaxb.model.v3.dev1.record.Addresses;
+import org.orcid.persistence.dao.AddressDao;
+import org.orcid.persistence.jpa.entities.AddressEntity;
+import org.springframework.cache.annotation.Cacheable;
+
+public class AddressManagerReadOnlyImpl extends ManagerReadOnlyBaseImpl implements AddressManagerReadOnly {
+    
+    @Resource
+    protected JpaJaxbAddressAdapter adapter;
+    
+    protected AddressDao addressDao;
+    
+    public void setAddressDao(AddressDao addressDao) {
+        this.addressDao = addressDao;
+    }
+
+    @Override
+    @Cacheable(value = "primary-address", key = "#orcid.concat('-').concat(#lastModified)")
```

Comment:
Hey @bobcaprice, we removed this annotation from the 2.0 manager, i think because we are using the cache at the DAO layer.

We need to decide with @wjrsimpson if we are going to leave caching to the DAO layer or move it all to the manager layer.